nodeJS/express/introduction_to_express.md

```diff
@@ -0,0 +1,98 @@
+
+```bash
+npm install express
+```
+
+We can now create an `app.js` file that will serve as the starting point for our Express server. We could name this file anything we want, such as `index.js` or `main.js`, but we will use the same name used in the Express documentation. Inside `app.js`, we will have the following:
+
+```javascript
+const express = require("express");
+const app = express();
+
+app.get("/", (req, res) => res.send("Hello, world!"));
+
+const PORT = 3000;
+app.listen(PORT, () => console.log(`My first Express app - listening on port ${PORT}!`));
+```
+
+Let's break this down. Firstly, we import `express` using `require()`, then we call it to create the actual server that we store in the `app` variable. While you can technically do this here in one line via `const app = require("express")();`, very often you will need features that come from properties on `express`. Many of these features will be covered in later lessons.
+
+We then have a `route handler` - the line beginning with `app.get`. We will come back to this in a moment.
+
+Finally, we tell our server to listen for requests on port 3000 of our [localhost](https://simple.wikipedia.org/wiki/Localhost) (which is basically just the computer's local connection). 3000 is an arbitrary and somewhat conventional default choice, but you can technically use any unused port (e.g. Vite's dev server uses port 5173 by default). Back in your terminal, if you enter `node app.js` then all being well, you should see `My first Express app - listening on port 3000!` logged. Congratulations! Your first Express server is now running.
+
+### A request's journey
+
+Now that our server is running and listening on port 3000, let's send it a request. In a browser, navigate to `http://localhost:3000/` (if you omit the `/` at the end, then the browser will invisibly add it for you). All this does is tell the browser to send a `GET` request to the `/` route of whatever server is at port 3000 of our localhost (which is our Express server!) and display in the window whatever it receives in response.
+
+Whenever you navigate to any web URL this way, this is essentially what you are doing. Navigating to `https://theodinproject.com/paths` via the address bar is just telling the browser to send a `GET` request to the `/paths` route at `https://theodinproject.com`, then display what it receives in response.
+
+Once you navigate to `http://localhost:3000/`, you should see `Hello, world!` appear in the window. Let's break down how our Express code got us here.
+
+When our server receives our `GET` request, Express stores the request in a [`request` object](https://expressjs.com/en/4x/api.html#req). This request gets passed through a chain of functions we call `middleware functions` until eventually, a middleware function tells Express to actually respond to the request.
```

Comment:
This is one of several places with a link where part of the link text is code formatted; this [tends to look somewhat wonky](https://www.theodinproject.com/lessons/preview?uuid=bd4a27c8-f851-4a97-ae83-1990be86e4a9) because the underline will get split between the code and the rest of the text, making it look like they are separate links

Reply:
Fair shout! Looking at it again, it would perfectly suffice without the backticks anyway given they're actually the only places I had those words backticked. The link visuals are sufficient.